bin/get-github-oauth-token

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env node
+
+var Github = require('github'),
+    read   = require('read');
+
+read({
+   prompt: "Github Username:"
+}, function(error, response) {
+   getPassword({username: response});
+})
+
+function getPassword(auth) {
+   read({
+      silent: true,
+      prompt: "Github Password:"
+   }, function(error, response) {
+      auth.password = response;
+      getOTP(auth);
+   })
+}
+
+function getOTP(auth) {
+   read({
+      prompt: "2-Factor Auth Code (leave blank for none):"
```

Comment:
I forgot some people use SMS for 2FA... This worked fine with `Google Authenticator`

> For users with two-factor authentication enabled, Basic Authentication requires an extra step. When you attempt to authenticate with Basic Authentication, the server will respond with a 401 and an X-GitHub-OTP: required; :2fa-type header. This indicates that a two-factor authentication code is needed (in addition to the username and password). The :2fa-type in this header indicates whether the account receives its two-factor authentication codes via SMS or via an application.

That doesn't say if they actually *send* the sms when you make that failed api call. Maybe this whole script should just be one echo statement:

> Go to https://github.com/settings/tokens and create a token with `repo:status` permission

Which is a bit simpler.